`realtime/auth.authbase0` test case can fail due to race condition

`authbase0` test case in `realtime/auth` can fail with the next error:

```
  1) realtime/auth
       authbase0:
     AssertionError: Verify token issued: expected false to be truthy
      at /mnt/c/Users/Andrey/dev/ably/ably-js/test/realtime/auth.test.js:76:105
      at /mnt/c/Users/Andrey/dev/ably/ably-js/test/common/modules/shared_helper.js:70:9
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

The error is caused by comparing `tokenDetails.issued` generated in `authbase0` using AblyRealtime with queryTime undefined (defaults to false) with `currentTime` from setup function generated via calling `AblyRest.time()` with `queryTime: true`.

Due to different `queryTime` setting used, if the local offset from server is large enough (set in `BaseClient.serverTimeOffset`) or the test case runs fast enough, we can retrieve a new token in `authbase0` before time offset ends, thus causing `tokenDetails.issued` to be less than saved `currentTime` and failing the test.



┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/ECO-4128) by [Unito](https://www.unito.io)

--- a/test/realtime/auth.test.js
+++ b/test/realtime/auth.test.js
@@ -64,7 +64,7 @@ define(['ably', 'shared_helper', 'async', 'chai'], function (Ably, helper, async
      * Base token generation case
      */
     it('authbase0', function (done) {
-      var realtime = helper.AblyRealtime();
+      var realtime = helper.AblyRealtime({ queryTime: true });
       whenPromiseSettles(realtime.auth.requestToken(), function (err, tokenDetails) {
         if (err) {
           closeAndFinish(done, realtime, err);

